bundles/binding/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/SoulissBinding.java

```diff
@@ -0,0 +1,326 @@
+		super.handleEvent(event);
+	}
+
+	@Override
+	/**
+	 * Get the souliss's typical from the hash table and send a command
+	 * @author Tonino Fazio
+	 * @since 1.7.0
+	 */
+	public void receiveCommand(String itemName, Command command) {
+
+		// Get the typical defined in the hash table
+		SoulissGenericTypical T = SoulissGenericBindingProvider.SoulissTypicalsRecipients
+				.getTypicalFromItem(itemName);
+		LOGGER.info("receiveCommand - " + itemName + " = " + command
+				+ " - Typical: " + T.getType());
+
+		switch (T.getType()) {
+		case Constants.Souliss_T11:
+			SoulissT11 T11 = (SoulissT11) T;
+			T11.CommandSEND(StateTraslator.commandsOHtoSOULISS(T.getType(),
+					command.toString()));
+			break;
+		case Constants.Souliss_T12:
+			SoulissT12 T12 = (SoulissT12) T;
+			T12.CommandSEND(StateTraslator.commandsOHtoSOULISS(T.getType(),
+					command.toString()));
+			break;
+		case Constants.Souliss_T14:
+
+			break;
+		case Constants.Souliss_T16:
+			SoulissT16 T16 = (SoulissT16) T;
+			String cmd = command.getClass().getSimpleName();
+			if (cmd.equals(Constants.Openhab_RGB_TYPE)) {
+				String HSB[] = command.toString().split(",");
+				short RGB[] = HSBtoRGB(Float.parseFloat(HSB[0]),
+						Float.parseFloat(HSB[1]), Float.parseFloat(HSB[2]));
+				T16.CommandSEND(StateTraslator.commandsOHtoSOULISS(T.getType(),
+						command.getClass().getSimpleName()), RGB[0], RGB[1],
+						RGB[2]);
+			} else
+				T16.CommandSEND(StateTraslator.commandsOHtoSOULISS(T.getType(),
+						command.toString()));
+			break;
+		case Constants.Souliss_T18:
+
+			break;
+		case Constants.Souliss_T19:
+			SoulissT19 T19 = (SoulissT19) T;
+			if (command instanceof PercentType) {
+				int percentToShort = (((PercentType) command).shortValue() * 255 / 100);
+				T19.CommandSEND(Constants.Souliss_T1n_Set,
+						Short.parseShort(String.valueOf(percentToShort)));
+			} else
+				T19.CommandSEND(StateTraslator.commandsOHtoSOULISS(T.getType(),
+						command.toString()));
+			break;
+		case Constants.Souliss_T21:
+
+			break;
+		case Constants.Souliss_T22:
+			SoulissT22 T22 = (SoulissT22) T;
+			T22.CommandSEND(StateTraslator.commandsOHtoSOULISS(T.getType(),
+					command.toString()));
+			break;
+		case Constants.Souliss_T_TemperatureSensor:
+			break;
+		case Constants.Souliss_T_HumiditySensor:
+			break;
+		case Constants.Souliss_T32_IrCom_AirCon:
+			break;
+		case Constants.Souliss_T41_Antitheft_Main:
+			break;
+		case Constants.Souliss_T42_Antitheft_Peer:
+			break;
+		default:
+			LOGGER.debug("Typical Unknown");
+		}
+	}
+
+	/**
+	 * Convert color format from HSB to RGB
+	 * 
+	 * @param H
+	 * @param S
+	 * @param B
+	 * @return short RGBList[] contain RGB components
+	 */
+	private short[] HSBtoRGB(Float H, Float S, Float B) {
+
+		int RGB = Color.HSBtoRGB(H, S, B);
+		Color c = new Color(RGB);
+		short RGBList[] = { (short) c.getRed(), (short) c.getGreen(),
+				(short) c.getBlue() };
+		return RGBList;
+	}
+
+	@Override
+	protected void execute() {
+	}
```

Comment:
since you didn't implement execute() rather extend AbstractBinding than AbstractActiveBinding